src/Facade/Services/PartsStorageTypeFacadeService.cs

```diff
@@ -0,0 +1,122 @@
+﻿namespace Linn.Stores2.Facade.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq.Expressions;
+    using System.Threading.Tasks;
+    using Linn.Common.Facade;
+    using Linn.Common.Persistence;
+    using Linn.Stores2.Domain.LinnApps;
+    using Linn.Stores2.Domain.LinnApps.Exceptions;
+    using Linn.Stores2.Domain.LinnApps.Parts;
+    using Linn.Stores2.Domain.LinnApps.Stock;
+    using Linn.Stores2.Facade.Common;
+    using Linn.Stores2.Resources.Parts;
+
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+    public class PartsStorageTypeFacadeService : AsyncFacadeService<PartsStorageType, PartsStorageTypeKey, PartsStorageTypeResource, PartsStorageTypeResource, PartsStorageTypeResource>
+    {
+        private readonly IRepository<Part, int> partRepository;
+
+        private readonly IRepository<StorageType, string> storageTypeRepository;
+
+        private readonly IRepository<PartsStorageType, PartsStorageTypeKey> partStorageTypeRepository;
+
+        public PartsStorageTypeFacadeService(
+            IRepository<PartsStorageType, PartsStorageTypeKey> partStorageTypeRepository,
+            ITransactionManager transactionManager,
+            IBuilder<PartsStorageType> resourceBuilder,
+            IRepository<Part, int> partRepository,
+            IRepository<StorageType, string> storageTypeRepository)
+            : base(partStorageTypeRepository, transactionManager, resourceBuilder)
+        {
+            this.partRepository = partRepository;
+            this.storageTypeRepository = storageTypeRepository;
+            this.partStorageTypeRepository = partStorageTypeRepository;
+        }
+
+        protected override async Task<PartsStorageType> CreateFromResourceAsync(
+            PartsStorageTypeResource resource,
+            IEnumerable<string> privileges = null)
+        {
+            var part = await this.partRepository.FindByAsync(p => p.PartNumber == resource.PartNumber);
+
+            if (part == null || string.IsNullOrEmpty(resource.PartNumber))
+            {
+                throw new PartsStorageTypeException("Part Number is empty or doesn't exist!");
+            }
+
+            var storageType = await this.storageTypeRepository.FindByAsync(st => st.StorageTypeCode == resource.StorageTypeCode);
+
+            if (storageType == null || string.IsNullOrEmpty(resource.StorageTypeCode))
+            {
+                throw new PartsStorageTypeException("Storage Type is empty or doesn't exist!");
+            }
```

Comment:
I would move both these checks down into your domain somewhere (since arguably 'Part Storage Types must have a part'  and 'Part StorageTypes musy have a storage type' are domain rules?) Then you could unit test them too